packages/video-player/src/VideoPlayer.tsx

```diff
@@ -1,31 +1,52 @@
 import { flattenStyles } from "@native-mobile-resources/util-widgets";
 import { Component, createElement } from "react";
-import { ActivityIndicator, View } from "react-native";
-import Video from "react-native-video";
-
+import { ActivityIndicator, Text, View } from "react-native";
+import Video, { OnLoadData } from "react-native-video";
 import { VideoPlayerProps } from "../typings/VideoPlayerProps";
 import { defaultVideoStyle } from "./ui/Styles";
 
 export type Props = VideoPlayerProps<undefined>;
 
+const enum StatusEnum {
+    ERROR = "error",
+    LOADING = "loading",
+    READY = "ready",
+    NOT_READY = "not-ready"
+}
+
 interface State {
-    loading: boolean;
+    aspectRatio?: number;
+    status: StatusEnum;
 }
 
 export class VideoPlayer extends Component<Props, State> {
     readonly state = {
-        loading: true
+        aspectRatio: undefined,
+        status: StatusEnum.NOT_READY
     };
 
     private readonly onLoadStartHandler = this.onLoadStart.bind(this);
     private readonly onLoadHandler = this.onLoad.bind(this);
+    private readonly onErrorHandler = this.onError.bind(this);
     private readonly styles = flattenStyles(defaultVideoStyle, this.props.style);
 
     render(): JSX.Element {
         const uri = this.props.videoUrl && this.props.videoUrl.value;
+
+        const styles = { ...this.styles.container };
+
+        if (!!this.props.aspectRatio && this.state.aspectRatio) {
```

Comment:
The `!!` doesn't do anything because it's already a boolean, right?
```suggestion
        if (this.props.aspectRatio && this.state.aspectRatio) {
```